kano_settings/system/advanced.py

```diff
@@ -282,6 +285,10 @@ def set_hosts_blacklist(enable, block_search,
     if enable:
         logger.debug('enabling blacklist')
 
+        if os.path.exists(hosts_file_backup):
+            logger.debug('restoring original backup file')
+            shutil.copy(hosts_file_backup, hosts_file)
```

Comment:
The backup file should be small, as it doesn't contain the parental control stuff.
Good catch re copy/copyfile: Apparently copy propagates the permissions, copyfile does not.  I think we should probably use copy when making the backup as well as when restoring it, so the permissions are propagated both ways, I'll add a change to do that....